### PR TITLE
new device: Juniper Wireless (Trapeze)

### PR DESCRIPTION
**Pull request by sirHulda**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

**Comment by laf** [comment]
I've not checked code but you need to sign the contributors agreement using the text we provide in the docs. Please add another commit to this PR with the required text.

File: includes/definitions.inc.php [hunk 1]
```diff
@@ -2228,6 +2228,11 @@
 $config['os'][$os]['over'][2]['graph'] = 'device_mempool';
 $config['os'][$os]['over'][2]['text']  = 'Memory Usage';
 
+$os = 'trapeze';
+$config['os'][$os]['text']             = 'Juniper Wireless';
+$config['os'][$os]['type']             = 'wireless';
+$config['os'][$os]['icon']             = 'junos';
+
```
Comment (laf) [inline includes/definitions.inc.php]: Can we not show any over graphs here? Look at some of the other OS definitions.

File: includes/discovery/os/trapeze.inc.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
+//
+// Discovery of Juniper Wireless (Trapeze) devices.
+//
+if (!$os) {
+    if (preg_match('/^Juniper Networks/', $sysDescr)) {
+        if (str_contains($sysDescr, 'MX-')) {
```
Comment (laf) [inline includes/discovery/os/trapeze.inc.php]: How sure are we that MX- would only be for trapeze devices?

File: includes/definitions.inc.php [hunk 3]
```diff
@@ -2228,6 +2228,11 @@
 $config['os'][$os]['over'][2]['graph'] = 'device_mempool';
 $config['os'][$os]['over'][2]['text']  = 'Memory Usage';
 
+$os = 'trapeze';
```
Comment (laf) [inline includes/definitions.inc.php]: Is trapeze the right name? We base it on firmware / os normally.

File: includes/discovery/os/trapeze.inc.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+            $os = 'trapeze';
+        }
+        if (str_contains($sysDescr, 'WLC-')) {
```
Comment (laf) [inline includes/discovery/os/trapeze.inc.php]: str_contains can accept arrays so you may as well put `array('MX-', 'WLC-')` into the first `str_contains` check.

File: includes/discovery/processors/trapeze.inc.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?php
+//
+// Discovery of CPU usage on Juniper Wireless (Trapeze) devices.
+//
+if ($device['os'] == 'trapeze') {
+    $descr = 'Processor';
+    $usage = snmp_get($device, 'trpzSysCpuLastMinuteLoad.0', '-OQUvs', 'TRAPEZE-NETWORKS-SYSTEM-MIB', 'trapeze');
+
+    if (is_numeric($usage)) {
+        discover_processor($valid['processor'], $device, '1.3.6.1.4.1.14525.4.8.1.1.11.2.0', '0', 'trapeze', $descr, '1', $usage, null, null);
```
Comment (laf) [inline includes/discovery/processors/trapeze.inc.php]: Needs a . at the start of the oid like `.1.3.6....`

File: includes/discovery/sensors/states/trapeze.inc.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        TrpzSysPowerSupplyStatus
+            INTEGER {
+                    other        (1),
+                    unknown      (2),
+                    ac-failed    (3),
+                    dc-failed    (4),
+                    ac-ok-dc-ok  (5)
+                }
+
+        The LibreNMS generic states is derived from Nagios:
```
Comment (laf) [inline includes/discovery/sensors/states/trapeze.inc.php]: Can get rid of the comments about nagios derived status, we have that in the docs.

File: includes/polling/os/trapeze.inc.php [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?php
+//
+// Poll information of Juniper Wireless (Trapeze) devices.
+//
+if ($device['os'] == 'trapeze') {
+    list(,,,$hardware,$version,) = explode(' ', $poll_device['sysDescr']);
+    $serial = snmp_get($device, 'trpzSerialNumber.0', '-OQv', 'TRAPEZE-NETWORKS-BASIC-MIB', 'trapeze');
```
Comment (laf) [inline includes/polling/os/trapeze.inc.php]: Could wrap these three calls into `snmp_get_multi()` to save exec time.

**Comment by laf** [comment]
I've put some inline comments in - needs the contributors agreement doing as per previous comment @sirHulda

File: includes/definitions.inc.php (hunk 1 above)
Comment (sirHulda) [inline includes/definitions.inc.php]: Sure. But i thought about adding it later on as i have plans to add more sensors such as sessions counters for each AP.

File: includes/discovery/os/trapeze.inc.php (hunk 2 above)
Comment (sirHulda) [inline includes/discovery/os/trapeze.inc.php]: Juniper also have MX routers but they are named "MX50" or "MX240" without "-".
I will have a look at it and see if there is other ways to determine if it's a Trapeze/Juniper Wireless device.

File: includes/definitions.inc.php (hunk 3 above)
Comment (laf) [inline includes/definitions.inc.php]: That's fine as long as trapeze is the os name, the firmware name or if neither of those is what they are known as. We try and avoid using hardware / vendor names as if they release multiple device types (say routers) then we can't have the same name for both types.

File: includes/definitions.inc.php (hunk 1 above)
Comment (laf) [inline includes/definitions.inc.php]: Other options can be added / changed later but it's best to have at least some set by default such as traffic, mem and cpu

**Comment by sirHulda** [comment]
Will do an overhaul and separate Trapeze(MX) and Juniper(WLC) devices as they have almost identical, but still different, mib sets. I will also create extra graphs and sensors for the controller and APs.
This might take some time so i've decided to close this pull request and create a new when i'm finished

**Comment by laf** [comment]
If they run the same os/firmware and are both witless then you don't need to separate out. 

Worth looking at our using git doc btw so you don't work in your master branch :)

**Pull request closed without merging**